discovery_engine/test/repository/hive_source_preferences_repo_test.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:io';
+
+import 'package:hive/hive.dart' show Hive;
+import 'package:test/test.dart';
+import 'package:xayn_discovery_engine/src/domain/event_handler.dart';
+import 'package:xayn_discovery_engine/src/domain/models/source.dart'
+    show Source;
+import 'package:xayn_discovery_engine/src/domain/models/source_preference.dart';
+import 'package:xayn_discovery_engine/src/domain/repository/source_preference_repo.dart';
+import 'package:xayn_discovery_engine/src/infrastructure/repository/hive_source_preference_repo.dart';
+
+Future<void> main() async {
+  group('HiveSourceFiltersRepository', () {
+    late SourcePreferenceRepository repo;
+
+    final sources = {
+      SourcePreference(Source('example.de'), PreferenceMode.favorite),
+      SourcePreference(Source('nytimes.com'), PreferenceMode.excluded),
+      SourcePreference(Source('example.com'), PreferenceMode.excluded),
+      SourcePreference(Source('include.com'), PreferenceMode.favorite),
+    };
+
+    setUpAll(() async {
+      EventHandler.registerHiveAdapters();
+    });
+
+    setUp(() async {
+      final dir = Directory.systemTemp.createTempSync('hive-test');
+      await EventHandler.initDatabase(dir.path);
+      repo = HiveSourcePreferenceRepository();
+    });
+
+    tearDown(() async {
+      await Hive.deleteFromDisk();
+    });
+
+    test('works when the box is empty', () async {
+      expect(await repo.getFavorites(), equals(<String>{}));
+      expect(await repo.getFavorites(), equals(<String>{}));
+    });
+
+    test('allows to retrieve source preferences', () async {
+      for (final el in sources) {
+        await repo.save(el);
+      }
+
+      final favorites = (await repo.getFavorites()).map((e) => e.toString());
```

Comment:
```suggestion
      final favorites = (await repo.getFavorites()).map((e) => e.value);
```

Reply:
That doesn't compile for me 🤔 Where would this value getter be coming from, I don't see it in the code

Reply:
https://github.com/xaynetwork/xayn_discovery_engine/blob/main/discovery_engine/lib/src/domain/models/source.dart#L39

Reply:
maybe you need to rebase :)